include/common/logger.h

```diff
@@ -1 +1,55 @@
-class logger{};
\ No newline at end of file
+#pragma once
+#include <string>
+#include <fstream>
+#include <ctime>
+#include <chrono> 
+
+class Logger {
+public:
+	enum loglevel
+	{
+		logCRITICAL, logERROR, logWARNING, logDEBUG, logINFO,
+	};
+	Logger(std::string file_name, loglevel level) : full_buffer_size (100)
+	{
+		log_level_needed = level;
```

Comment:
initialize log_level_needed in intializer list(like you did with full_buffer_size above